### PR TITLE
`make dist` as singular dependency for downstream distributors

### PR DESCRIPTION
**Pull request by dickmao**
rpm and debian maintainers should only ever run `make dist` from their
clone of the upstream repository.  ESS maintainers should only
need to ensure `make dist` works (and produces a complete set of artifacts).

@dsteuer We need to know what `make` targets you use from our toplevel Makefile, and your `rpmbuild` incantation, generally.  You speak of a "good old spec file" in #731 .

@eddelbuettel We have good visibility into the debian build.  An unnecessary but "would be nice" modification in https://salsa.debian.org/edd/ess/merge_requests/1/diffs

Subsumes #707

### CODE REVIEW

**Comment by eddelbuettel** [comment]
Actually, I tend to work off the tar.gz and I would be happy as a clam if that tar.gz were self-sufficient.  As I said before, I have reason to believe it is currently not self-sufficient what we can take those Debian aspect up over there.

**Comment by dickmao** [comment]
Ah, using the tarball is something of a showstopper for this PR.
You don't use `gbp buildpackage`?  That seemed to work for me, but I don't have your experience.

**Comment by eddelbuettel** [comment]
The work flow typically consists of importing a new upstream release, commonly a compressed tar file, into its branch first (most often via `gbp import-orig --uscan` which utilizes the `debian/watch` files and the `uscan` tool to locate and download the tarball) and to then build off the updated repo.  I do call `gbp buildpackage` but it also calls into `pbuilder` to have a "clean" and "pristine" empty machine alike (`pbuilder` is like an adapted `chroot` system from the pre-Docker days). It's been a while but I think that is a one-time setup to declare the (already existing on my box) `pbuilder` inside the `~/.gbp.conf`.   Maybe we should have that conversation over at salsa, it may not have too much relevance for anybody else here.  Appreciate you digging into it though.

File: Makefile [hunk 1]
```diff
@@ -44,75 +44,11 @@ julia:
 autoloads:
 	cd lisp; $(MAKE) ess-autoloads.el
 
-## the rest of the targets are for ESS developer's use only :
-
-## --- PRE-release ---
-
-# Create .tgz and .zip files only
-# GNUTAR=gtar make tarballs
-tarballs: $(ESSDIR)
```
Comment (jabranham) [inline Makefile]: Why remove the tarballs target?

File: Makefile [hunk 2]
```diff
@@ -30,7 +30,7 @@ etc: version
 	cd etc; $(MAKE)
 
 .PHONY: test
-test: version
+test: version julia
```
Comment (jabranham) [inline Makefile]: Why does julia get added here? AFAIR we don't test julia at all

**Comment by dickmao** [comment]
> The work flow typically consists of importing a new upstream release, commonly a compressed tar file, into its branch first (most often via `gbp import-orig --uscan` which utilizes the `debian/watch` files

Okay, I'm unaware of a lot.  The `debian/watch` points to http://ess.r-project.org/downloads/ess which contains tarballs in all their resplendence (I excised the tarball making thinking they were largely historical).  I need to close this PR.

**Comment by mmaechler** [comment]
> Okay, I'm unaware of a lot. ............ I need to close this PR.

Indeed, thank you.

For future reference:  The *main* release targets are the tarballs (now with their signature files) and the `make rel` target includes "uploading" them to the download area of  https://ess.r-project.org/
and to automagically update those web pages.

Even if that target is only used by me currently, we have put quite a bit of effort into it, if only to ensure nothing is forgotten.  (@jabranham and we still must fix something there such that git `tag` is not just pushed but specifically "pushed into master" (something I really don't quite get - one of the wonders of git ..).

**Pull request closed without merging**